Engine/envvar.c

```diff
@@ -1046,6 +1047,11 @@ void *csoundFileOpenWithType(CSOUND *csound, void *fd, int type,
         }
         fullName = (char*) name;
       }
+      free(wfname);
```

Comment:
Because of the goto in line 1045, it's possible for wfname and wmode to leak memory.

Reply:
Oh yeah oops. Would it be good enough to just copy the two free lines immediately before the goto?

Reply:
Also, is there a way to add a test for the memory leak?

Reply:
Ok, copied the frees before the goto